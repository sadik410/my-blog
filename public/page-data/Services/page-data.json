{"componentChunkName":"component---src-pages-services-tsx","path":"/Services/","result":{"data":{"site":{"siteMetadata":{"services":[{"icon":" fa fa-globe ","title":"Web Development","subtitle":"we develop web applications with react JS"},{"icon":" fa fa-mobile ","title":"Mobile Development","subtitle":"we develop mobile applications with react-native & react-native-web"},{"icon":" fa fa-desktop ","title":"Web Design","subtitle":"we do the design for web and mobile applications"}]}}},"pageContext":{}},"staticQueryHashes":["1717071199","374561337"]}