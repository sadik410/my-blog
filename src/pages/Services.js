import React from "react";

import Layout from "../components/layout";
import SEO from "../components/seo";
import Section from "../components/section/Section";
import ServiceItem from "../components/services/serviceItem";

const services = [
  {
    icon: " fa fa-globe ",

    title: "Web Development",
    subtitle: "we develop web applications with react JS",
  },
  {
    icon: " fa fa-mobile ",
    title: "Mobile Development",
    subtitle:
      " we develop mobile applications with react-native & react-native-web",
  },
  {
    icon: " fa fa-desktop ",
    title: "Web Design",
    subtitle: "we do the design for web and mobile applications",
  },
];
const Services = () => (
  <Layout>
    <SEO title="Services" />
    <Section title="Services">
      <div className="container services">
        <h4 className="title">
          My
          <span
            style={{
              color: "#2196f3",
            }}
          >
            {" Services"}
          </span>
        </h4>
        <div className="row">
          {services.map(({ icon, title, subtitle }) => (
            <ServiceItem
              key={title}
              icon={icon}
              title={title}
              subtitle={subtitle}
            />
          ))}
        </div>
      </div>
    </Section>
  </Layout>
);

export default Services;
