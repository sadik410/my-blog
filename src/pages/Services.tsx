import * as React from "react";
import { graphql } from "gatsby";

import Layout from "../components/layout";
import SEO from "../components/seo";
import Section from "../components/section/Section";
import ServiceItem from "../components/services/serviceItem";
import { myContext } from "../context/Provider";

const Services = ({ data }) => {
  const { activeColor } = React.useContext(myContext);
  return (
    <Layout>
      {/*       <SEO title="Services" />
       */}
      <Section title="Services">
        <div className="container services">
          <h4 className="title">
            My
            <span
              style={{
                color: activeColor,
              }}
            >
              {" Services"}
            </span>
          </h4>
          <div className="row">
            {data?.site?.siteMetadata?.services.map(
              ({ icon, title, subtitle }) => (
                <ServiceItem
                  key={title}
                  icon={icon}
                  title={title}
                  subtitle={subtitle}
                />
              )
            )}
          </div>
        </div>
      </Section>
    </Layout>
  );
};

export default Services;

export const query = graphql`
  query ServicesPageQuery {
    site {
      siteMetadata {
        services {
          icon
          title
          subtitle
        }
      }
    }
  }
`;
