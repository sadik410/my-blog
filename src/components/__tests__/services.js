import * as React from "react";
import { render, screen } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Services from "../../pages/Services";

const services = [
  {
    icon: " fa fa-globe ",

    title: "Web Development",
    subtitle: "we develop web applications with react JS",
  },
  {
    icon: " fa fa-mobile ",
    title: "Mobile Development",
    subtitle:
      "we develop mobile applications with react-native & react-native-web",
  },
  {
    icon: " fa fa-desktop ",
    title: "Web Design",
    subtitle: "we do the design for web and mobile applications",
  },
];
beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => ({
    siteMetadata: {
      title: `Gatsby Default Starter`,
      description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
      author: `@gatsbyjs`,
      services,
    },
  }));
});

afterAll(() => {
  useStaticQuery.mockRestore();
});

describe("Home", () => {
  it("renders correctly", () => {
    const { container } = render(<Services />);
    expect(container).toMatchSnapshot();
  });
  it("render My Services correctly ", () => {
    const { getByText } = render(<Services />);
    expect(
      getByText((content, element) => {
        return (
          element.tagName.toLowerCase() === "h4" &&
          element.textContent === "My Services"
        );
      })
    ).toBeInTheDocument();
  });

  /* it("render My Services correctly ", () => {
    const { getByText } = render(<Services />);
    screen.debug();
    expect(
      getByText((content, element) => {
        return (
          element.tagName.toLowerCase() === "h4" &&
          element.textContent === "Web Development"
        );
      })
    ).toBeInTheDocument();
  }); */
  /* it("render role correctly ", () => {
    const { getByText } = render(<Services />);
    mySkills.forEach((item) => {
      const getTagLi = (content, element) => {
        return element.tagName.toLowerCase() === "li";
      };
      expect(getByText(getTagLi)).toBeInTheDocument();
    });
  }); */
});
