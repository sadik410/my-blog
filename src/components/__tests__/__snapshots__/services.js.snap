// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="panel"
    >
      <button
        class="panel__switcher"
      >
        <i
          aria-hidden="true"
          class="fa fa-gear"
        />
      </button>
      <div
        class="panel__body "
      >
        <div
          class="panel__title"
        >
          Main Color
        </div>
        #2196F3
        <div
          class="panel__main-color d-flex justify-content-between flex-wrap"
        />
      </div>
    </div>
    <header
      style="padding-top: 20px; padding-bottom: 20px;"
    >
      <div
        class="header row justify-content-between"
      >
        <h1
          class="name"
        >
          Sadik
          <span
            style="color: rgb(33, 150, 243);"
          >
             Milliti
          </span>
        </h1>
        <nav>
          <ul
            class=""
          />
        </nav>
      </div>
    </header>
    <div>
      <main>
        <section
          class=" flex-column section"
        >
          <div
            class="section__header"
            style="background-color: rgb(33, 150, 243);"
          >
            <h2
              class="section__title"
            >
              Services
            </h2>
          </div>
          <div
            class="section__body"
          >
            <div
              class="container services"
            >
              <h4
                class="title"
              >
                My
                <span
                  style="color: rgb(33, 150, 243);"
                >
                   Services
                </span>
              </h4>
              <div
                class="row"
              />
            </div>
          </div>
        </section>
      </main>
      <footer
        style="margin-top: 2rem;"
      >
        Â© 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.com"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>
</div>
`;
